### PR TITLE
Remove call_without_output from Shell.py L7

### LINKED ISSUE #3600: Remove call_without_output from Shell.py L7
This line was used by the requirement classes, it isnt used anymore as they use sarge, so it should be removed.
difficulty/newcomer

### PR DESCRIPTION
**Pull request by SiddharthKumar02**
closes #3600

### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [x] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

File: coalib/misc/Shell.py [hunk 1]
```diff
@@ -1,10 +1,7 @@
...
-import functools
 import shlex
-from subprocess import PIPE, Popen, call, DEVNULL
-
-
-call_without_output = functools.partial(call, stdout=DEVNULL, stderr=DEVNULL)
+from contextlib import contextmanager
+from subprocess import PIPE
+from subprocess import Popen
 """
```
Comment (Adrianzatreanu) [inline coalib/misc/Shell.py]: why did you remove the function but kept the docstring?

File: coalib/misc/Shell.py [hunk 2]
```diff
@@ -1,10 +1,7 @@
-from contextlib import contextmanager
-import functools
 import shlex
-from subprocess import PIPE, Popen, call, DEVNULL
-
-
-call_without_output = functools.partial(call, stdout=DEVNULL, stderr=DEVNULL)
+from contextlib import contextmanager
+from subprocess import PIPE
+from subprocess import Popen
```
Comment (Adrianzatreanu) [inline coalib/misc/Shell.py]: why did you change the imports? i think its totally unrelated to the issue..

**Comment by Adrianzatreanu** [comment]
unack 4516370e0844a0fcc80ab0611120f4cf57dd120c

File: coalib/misc/Shell.py (hunk 2 above)
Comment (SiddharthKumar02) [inline coalib/misc/Shell.py]: The gitmate-bot asked me to.

Comment (nemaniarjun) [inline coalib/misc/Shell.py]: I think It's unused code. That's why the error.

Comment (SiddharthKumar02) [inline coalib/misc/Shell.py]: @nemaniarjun Yeah precisely that. So is the code fine then? Because it did clear all the tests.

File: coalib/misc/Shell.py (hunk 1 above)
Comment (SiddharthKumar02) [inline coalib/misc/Shell.py]: I didn't pay attention to the docstring, sorry about that. :/

File: coalib/misc/Shell.py [hunk 3]
```diff
@@ -1,14 +1,7 @@
...
 import shlex
-from subprocess import PIPE, Popen, call, DEVNULL
-
-
-call_without_output = functools.partial(call, stdout=DEVNULL, stderr=DEVNULL)
-"""
-Uses subprocess.call to execute a command, but suppresses the output and
-the errors.
-"""
+from contextlib import contextmanager
```
Comment (Mixih) [inline coalib/misc/Shell.py]: hmm this reorder should not have been needed. Can you try reverting it(i.e move the context manager one up)?

**Comment by Mixih** [comment]
unack 3694dd0

File: coalib/misc/Shell.py (hunk 3 above)
Comment (SiddharthKumar02) [inline coalib/misc/Shell.py]: Sure

**Comment by SiddharthKumar02** [comment]
@Mixih Is this good for a merge?

**Comment by supergr35** [comment]
Please include a commit message outlining the reasoning behind removing these lines.

**Comment by SiddharthKumar02** [comment]
@supergr35 Is this alright?

**Comment by affanv14** [comment]
ack d1cff3c

**Comment by supergr35** [comment]
ack d1cff3c

**Comment by Udayan12167** [comment]
unack d1cff3c

**Comment by SiddharthKumar02** [comment]
@Udayan12167 , What other changes do i have to make?

**Comment by SiddharthKumar02** [comment]
@Udayan12167 , Is this ok?

**Comment by gkrishnan724** [comment]
ack  	a05ec8c

**Comment by Udayan12167** [comment]
ack a05ec8c

**Comment by supergr35** [comment]
ack a05ec8c

**Comment by supergr35** [comment]
In your next PR, I highly recommend checking out a new branch to use for your development rather than working on master. You can find instructions here :) http://api.coala.io/en/latest/Developers/Git_Basics.html#creating-a-new-branch

**Comment by Udayan12167** [comment]
@rultor merge
